mtp_api/apps/performance/models.py

```diff
@@ -32,6 +33,34 @@ def digital_takeup_per_day(self):
                 'digital_takeup_per_day': mean(value['digital_takeup'] for value in group),
             }
 
+    def digital_takeup_per_month(self):
+        """
+        Per-month digital take-up averages
+        NB: Starts on 1/1/2017 because the data prior to that is poor
+        :return:  generator
+        """
+        with connection.cursor() as cursor:
+            cursor.execute('''
+                WITH credit_count AS (
+                  SELECT date_trunc('month', received_at) AS date,
+                    COUNT(*) AS accurate_credits_by_mtp
+                  FROM credit_credit
+                  WHERE resolution = 'credited' AND received_at >= %(min_date)s
+                  GROUP BY date_trunc('month', received_at)
+                ), average_takeup AS (
+                  SELECT date_trunc('month', date) AS date,
+                    SUM(credits_by_post) AS reported_credits_by_post,
+                    SUM(credits_by_mtp) AS reported_credits_by_mtp
+                  FROM performance_digitaltakeup
+                  WHERE date >= %(min_date)s
+                  GROUP BY date_trunc('month', date)
+                )
+                SELECT credit_count.date, accurate_credits_by_mtp, reported_credits_by_post, reported_credits_by_mtp
+                FROM credit_count FULL OUTER JOIN average_takeup ON credit_count.date = average_takeup.date
+                ORDER BY credit_count.date
+            ''', params={'min_date': datetime.date(2017, 1, 1)})
```

Comment:
Is having this as the only minimum sustainable? Shouldn't it also be based on the period that you're looking at?